File: Data Structures & Algorithms/longest-common-subsequence/submission-21.cs
public class Solution {
    public int LongestCommonSubsequence(string text1, string text2) {
        if (text1.Length < text2.Length) {
            string temp = text1;
            text1 = text2;
            text2 = temp;
        }

        int[] prev = new int[text2.Length + 1];
        int[] curr = new int[text2.Length + 1];

        for (int i = text1.Length - 1; i >= 0; i--) {
            for (int j = text2.Length - 1; j >= 0; j--) {
                if (text1[i] == text2[j]) {
                    curr[j] = 1 + prev[j + 1];
                } else {
                    curr[j] = Math.Max(curr[j + 1], prev[j]);
                }
            }
            Array.Copy(curr, prev, text2.Length + 1);
        }

        return prev[0];
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-21.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int result = 0;
        int l = 0, r = 1;

        while (r < prices.size()) {
            result = max(result, prices[r] - prices[l]);

            if (prices[l] > prices[r]) {
                l = r;
            }
            
            r++;
        }
        return result;
    }
};